cpp/leaky_core.cc

```diff
@@ -179,12 +181,64 @@ bool InputAvailable(int fd) {
   NotImplemented();
 }
 
+SignalHandler::SignalHandler()
+    : signal_queue(), last_sig_num(0), sigwinch_num(UNTRAPPED_SIGWINCH) {
+}
+
+void SignalHandler::EnqueueSignal(int sig_num) {
```

Comment:
Might want to call this `UpdateVars()`  or something, because it does more than enqueue a signal

That's why I was thinking `SignalSafe::Update()` or `SignalShared::Update()` are more descriptive

i.e. this class isn't really a signal handler -- it's a bunch of state that gets mutated by a signal handler!